### PR TITLE
test: replace testcontainers with spring-kafka EmbeddedKafka

### LINKED ISSUE #62: Replace testcontainers with "something else"?
I think the testcontainers tests are slow, and we will also be affected by the DockerHub rate limits. There are some suggestions in https://github.com/testcontainers/testcontainers-java/issues/3099, but I personally think we could/should test this without containers. What about an in-memory embedded broker?

### PR DESCRIPTION
**Pull request by erikgb**
This will downgrade our Kafka client version, which is unfortunate. I do not expect this PR to be approved/merged, but you could maybe take a look?

Close #62

### CODE REVIEW

**Comment by erikgb** [comment]
This is also relevant https://github.com/spring-projects/spring-kafka/issues/2391

**Comment by sverrehu** [comment]
Timing-wise the Spring approach an my approach are quite equal: https://github.com/statnett/k3a-lag-exporter/pull/65

**Comment by sverrehu** [comment]
Why do we have to downgrade the Kafka client jars to run this? I don't like the idea of making the version dependent on test code, especially when that pushes us three minor versions back in time.

File: src/test/java/no/statnett/k3alagexporter/itest/K3aLagExporterIT.java [hunk 1]
```diff
@@ -33,20 +35,14 @@ public final class K3aLagExporterIT {
     private int nextProducedValue = 0;
 
     @BeforeAll
-    public static void beforeClass() {
+    public static void beforeClass(EmbeddedKafkaBroker broker) {
```
Comment (sverrehu) [inline src/test/java/no/statnett/k3alagexporter/itest/K3aLagExporterIT.java:38]:
```suggestion
    public static void beforeClass(final EmbeddedKafkaBroker broker) {
```

For consistency.

File: src/test/java/no/statnett/k3alagexporter/itest/services/KafkaCluster.java [hunk 2]
```diff
@@ -10,28 +10,16 @@
...
-        container.start();
-    }
-
-    public void stop() {
-        container.stop();
-    }
-
-    public String getBootstrapServers() {
-        return String.format("%s:%s", container.getHost(), container.getMappedPort(KafkaContainer.KAFKA_PORT));
+    public KafkaCluster(EmbeddedKafkaBroker broker) {
```
Comment (sverrehu) [inline src/test/java/no/statnett/k3alagexporter/itest/services/KafkaCluster.java:21]:
```suggestion
    public KafkaCluster(final EmbeddedKafkaBroker broker) {
```

**Comment by erikgb** [comment]
> Why do we have to downgrade the Kafka client jars to run this? I don't like the idea of making the version dependent on test code, especially when that pushes us three minor versions back in time.

I totally agree, but it's Kafkas fault. It breaks if the versions are not in sync.

**Comment by sverrehu** [comment]
> I totally agree, but it's Kafkas fault. It breaks if the versions are not in sync.

Hm. In my test branch they will always be in sync, since the code for embedding is "embedded" in the project. Would prefer an external dependency, but if the server and client are so tightly knit, maybe this is the way to go.

**Comment by erikgb** [comment]
Closing in favor of https://github.com/statnett/k3a-lag-exporter/pull/65

**Pull request closed without merging**